drivers/modbus_rtu/include/modbus_rtu.h

```diff
@@ -0,0 +1,173 @@
+};
+
+//! Максимальный размер пакета протокола Modbus RTU.
+#define MODBUS_RTU_PACKET_SIZE_MAX 256
+
+//! Размер контрольной суммы протокола Modbus RTU.
+#define MODBUS_RTU_CRC_SIZE 2
+
+//! Размер полей сообщения протокола Modbus RTU.
+#define MODBUS_RTU_FIELDS_SIZE (sizeof(modbus_rtu_id_t) + sizeof(modbus_rtu_func_t))
+//! Размер полей и контрольной суммы сообщения протокола Modbus RTU.
+#define MODBUS_RTU_FIELDS_CRC_SIZE (MODBUS_RTU_CRC_SIZE + MODBUS_RTU_FIELDS_SIZE)
+//! Максимальный размер данных протокола Modbus RTU.
+#define MODBUS_RTU_DATA_SIZE_MAX (MODBUS_RTU_PACKET_SIZE_MAX - MODBUS_RTU_FIELDS_CRC_SIZE)
+//! Максимальный размер данных и контрольной суммы протокола Modbus RTU.
+#define MODBUS_RTU_DATA_CRC_SIZE_MAX (MODBUS_RTU_DATA_SIZE_MAX + MODBUS_RTU_CRC_SIZE)
+
+/**
+ * @brief Modbus message structure.
+ */
+typedef struct {
+  uint8_t id;     /**< ID of slave, on response not use */
+  uint8_t func;   /**< function code */
+  uint16_t addr;  /**< Starting Address */
+  uint16_t count; /**< Quantity of Registers or byte or bits */
+  uint16_t *regs; /**< Registers or byte or bits*/
+} modbus_rtu_message_t;
+
+/**
+ * @brief Modbus structure.
+ */
+typedef struct {
+  /**
+   * @brief  UART port.
+   */
+  uart_t uart;
+  /**
+   * @brief  baudrate.
+   */
+  uint32_t baudrate;
+  /**
+   * @brief  RTS pin. 0 - disable.
+   */
+  gpio_t pin_rts;
+  /**
+   * @brief  RTS signals when transmit.
+   */
+  int pin_rts_enable;
+  /**
+   * @brief  ID device. 0 - master, otherwice slave.
+   */
+  uint8_t id;
+  /**
+   * @brief   Before slave begin to response; usec.
+   */
+  uint32_t timeout;
+  /**
+   * @brief   Template buffer for receive message or send. **Must never be changed
+   *          by the user.**
+   * @internal
+   */
+  uint8_t _buffer[MODBUS_RTU_PACKET_SIZE_MAX];
+  /**
+   * @brief   Buffer size. **Must never be changed by the user.**
+   * @internal
+   */
+  uint8_t _size_buffer;
+  /**
+   * @brief   Mutex for block buffer. **Must never be changed by the user.**
+   * @internal
+   */
+  mutex_t _mutex_buffer;
+  /**
+   * @brief   Timeout beetwin dyte. **Must never be changed by the user.**
+   * @internal
+   */
+  uint32_t _rx_timeout;
+  /**
+   * @brief   pointer on current message. **Must never be changed by the user.**
+   * @internal
+   */
+  modbus_rtu_message_t *_msg;
+  /**
+   * @brief   PID thread. **Must never be changed by the user.**
+   * @internal
+   */
+  kernel_pid_t _pid;
+} modbus_rtu_t;
+
+/**
+ * @brief   Initialize the given modbus.
+ *
+ * Initialize fields of modbus and UART, RTS pin.
+ *
+ * @param[in] modbus    pointer modbus 
+ *
+ * @return              0 on success
+ * @return              less 0 on error
+ */
+int modbus_rtu_init(modbus_rtu_t *modbus);
+/**
+ * @brief   Send request to slave.
+ *
+ * Send request to slave, blocking and wait response.
+ *
+ * @param[in] modbus    pointer modbus 
+ * @param[in] message    pointer modbus message
```

Comment:
Nitpick: Please align the parameter and description. (Once you have configured your editor to use spaced over tabs, the tab key will be your friend.)